sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -376,6 +386,23 @@ object ScalaReflection extends ScalaReflection {
           dataType = ObjectType(udt.getClass))
         Invoke(obj, "deserialize", ObjectType(udt.userClass), getPath :: Nil)
 
+      case t if isValueClass(t) =>
+        // nested value class is treated as its underlying type
+        // top level value class must be treated as a product
+        val underlyingType = getUnderlyingTypeOf(t)
+        val underlyingClsName = getClassNameFromType(underlyingType)
+        val clsName = getUnerasedClassNameFromType(t)
+        val newTypePath = s"""- Scala value class: $clsName($underlyingClsName)""" +:
```

Comment:
do you have an example of this message?

Reply:
Sure, here is the message when running with value class `Id` above
`- Scala value class: org.apache.spark.sql.catalyst.encoders.Id(scala.Int)`